arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorHolder.java

```diff
@@ -91,17 +91,44 @@ public int numValues() {
     return vector.getValueCount();
   }
 
+  public static VectorHolder constantHolder(int numRows, Object constantValue) {
+    return new DummyVectorHolder(numRows, constantValue);
+  }
+
   public static VectorHolder dummyHolder(int numRows) {
-    return new VectorHolder() {
-      @Override
-      public int numValues() {
-        return numRows;
-      }
-    };
+    return new DummyVectorHolder(numRows);
   }
 
   public boolean isDummy() {
     return vector == null;
   }
 
+  /**
+   * A Vector Holder which does not actually produce values, consumers of this class should
+   * use the constantValue to populate their ColumnVector implementation.
+   */
+  public static class DummyVectorHolder extends VectorHolder {
```

Comment:
Could we rename this to `ConstantVectorHolder` instead of "dummy"? Now that it returns a constant value, I think it's more accurate to use "constant".

Reply:
Makes sense to me